tests/unit/abilities-api/wpAbilityCategory.php

```diff
@@ -0,0 +1,768 @@
+<?php declare( strict_types=1 );
+
+/**
+ * Tests for the ability category functionality.
+ *
+ * @covers WP_Ability_Category
+ * @covers WP_Abilities_Category_Registry
+ * @covers wp_register_ability_category
+ * @covers wp_unregister_ability_category
+ * @covers wp_get_ability_category
+ * @covers wp_get_ability_categories
+ *
+ * @group abilities-api
+ */
+class Tests_Abilities_API_WpAbilityCategory extends WP_UnitTestCase {
+
+	/**
+	 * Category registry instance.
+	 *
+	 * @var \WP_Abilities_Category_Registry
+	 */
+	private $registry;
+
+	/**
+	 * Captured `_doing_it_wrong` calls during a test.
+	 *
+	 * @var array<int,array{function:string,message:string,version:string}>
+	 */
+	private $doing_it_wrong_log = array();
+
+	/**
+	 * Set up before each test.
+	 */
+	public function set_up(): void {
+		parent::set_up();
+
+		$this->registry           = WP_Abilities_Category_Registry::get_instance();
+		$this->doing_it_wrong_log = array();
+
+		add_action( 'doing_it_wrong_run', array( $this, 'record_doing_it_wrong' ), 10, 3 );
+	}
+
+	/**
+	 * Tear down after each test.
+	 */
+	public function tear_down(): void {
+		remove_action( 'doing_it_wrong_run', array( $this, 'record_doing_it_wrong' ), 10 );
```

Comment:
Nit, 10 is the defualt priority:
```suggestion
		remove_action( 'doing_it_wrong_run', array( $this, 'record_doing_it_wrong' ) );
```

More broadly, would it make sense to set this action once for the class with `setUpBeforeClass` and remove it once with `tearDownAfterClass` at the end?

Reply:
Good catch on the priority parameter. Regarding class-level setup: we need per-test setup here because `setUpBeforeClass()` is static and can't access the instance method `record_doing_it_wrong()` or the instance property `$doing_it_wrong_log`.